src/poll.rs

```diff
@@ -53,7 +119,32 @@ pub fn poll(fds: &mut [PollFd], timeout: libc::c_int) -> Result<libc::c_int> {
     Errno::result(res)
 }
 
-#[cfg(any(target_os = "linux", target_os = "android"))]
+/// `ppoll()` allows an application to safely wait until either a file
+/// descriptor becomes ready or until a signal is caught.
+/// ([`poll(2)`](http://man7.org/linux/man-pages/man2/poll.2.html))
+///
+/// The operation of `ppoll` is identical to `poll`, other than these two differences:
+///
+/// * `timeout` is a `TimeSpec` which can specify time with nanosecond precision.
+/// * `ppoll` changes the signal set whose delivery are blocked for the caller before polling
+///    the file descriptors. This set is reverted after the polling.
+///    Note that this is done **atomically**.
+///
+/// It is equivalent to *atomically* executing the following calls
+/// (expressed here in C):
+///
+/// ```c
+/// sigset_t origmask;
+/// int timeout;
+///
+/// timeout = (tmo_p == NULL) ? -1 :
+///           (tmo_p->tv_sec * 1000 + tmo_p->tv_nsec / 1000000);
+/// pthread_sigmask(SIG_SETMASK, &sigmask, &origmask);
+/// ready = poll(&fds, nfds, timeout);
+/// pthread_sigmask(SIG_SETMASK, &origmask, NULL);
+/// ```
+///
+#[cfg(any(target_os = "linux", target_os = "android", target_os = "freebsd", target_os = "dragonfly"))]
```

Comment:
Can you alphabetize the list of target OSes here since you're already modifying it?